deepchecks/nlp/text_data.py

```diff
@@ -65,210 +62,116 @@ class TextData:
     task_type : str, default: None
         The task type for the text data. Can be either 'text_classification' or 'token_classification'. Must be set if
         label is provided.
-    dataset_name : t.Optional[str] , default: None
+    name : t.Optional[str] , default: None
         The name of the dataset. If None, the dataset name will be defined when running it within a check.
-    index : t.Optional[t.Sequence[int]] , default: None
-        The index of the samples. If None, the index is set to np.arange(len(raw_text)).
```

Comment:
So you're getting rid of the index?
Why?

Reply:
It is a cause of confusion and over complication, if you have index you have to handle cases that some of the input parameter come with different index, index not in correct order between inputs etc .. 
This way as a first step we keep things simple - samples are in-order across all inputs (tokenized text, text, label, properties, etc..)

Reply:
@noamzbr approved of this? Because the index was important to him

Reply:
yes